pymc_marketing/mmm/additive_effect.py

```diff
@@ -131,35 +132,31 @@ def model(self) -> pm.Model:
         """The PyMC model."""
 
 
-class MuEffect(Protocol):
-    """Protocol for arbitrary additive mu effect."""
+class MuEffect(ABC, BaseModel):
+    """Abstract base class for arbitrary additive mu effects.
 
+    All mu_effects must inherit from this Pydantic BaseModel to ensure proper
+    serialization and deserialization when saving/loading MMM models.
+    """
+
+    @abstractmethod
     def create_data(self, mmm: Model) -> None:
         """Create the required data in the model."""
 
+    @abstractmethod
     def create_effect(self, mmm: Model) -> pt.TensorVariable:
         """Create the additive effect in the model."""
 
+    @abstractmethod
     def set_data(self, mmm: Model, model: pm.Model, X: xr.Dataset) -> None:
         """Set the data for new predictions."""
 
 
-class FourierEffect:
+class FourierEffect(MuEffect):
     """Fourier seasonality additive effect for MMM."""
 
-    def __init__(self, fourier: FourierBase, date_dim_name: str = "date"):
-        """Initialize the Fourier effect.
-
-        Parameters
-        ----------
-        fourier : FourierBase
-            The FourierBase instance to use for the effect.
-        date_dim_name : str, optional
-            The name of the date dimension in the model, by default "date".
-
-        """
-        self.fourier = fourier
-        self.date_dim_name: str = date_dim_name
+    fourier: InstanceOf[FourierBase]
+    date_dim_name: str = Field("date")
```

Comment:
Do we need to supply the `date_dim_name` here, as the mutltidimensional MMM has the `date_column` attribute?